### ISSUE TITLE
WinActivate occasionally stalls, and program stops

### ISSUE THREAD

**Issue by Nickiel12** [spyoungtech/ahk#58]
I have a program that does a lot of window juggling, and occasionally the ahk script doesn't exit and my gui freezes because winactivate hasn't returned.

**Comment by spyoungtech** [comment]
Interesting. I know that with AutoHotkey, there is sometimes problems caused by rapidly switching windows one after the other, but as far as I none that should stall or block AHK from exiting.

I'll try replicating this behavior some time this week.

Some clarifying questions that may help get to the bottom of this:

* Are the windows you're trying to activate minimized?
* Are you using remote desktop?
* How many windows are you juggling and how quickly are you switching between them?


In short, I suspect this is an AHK issue more so than an issue with the wrapper. However, the wrapper may be able to provide workarounds for this if we can discover the root cause. One initial thought may be to specify a timeout that is passed to `subprocess.run` or put a time delay in your script between activating windows.

**Comment by spyoungtech** [comment]
Since there's not been any activity on this in a while, I'm going to close it for now. Feel free to reopen if you still need assistance.
